sdk-extension/opentelemetry-sdk-extension-aws/src/opentelemetry/sdk/extension/aws/trace/propagation/aws_xray_format.py

```diff
@@ -0,0 +1,284 @@
+
+    See: https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader
+    """
+
+    # AWS
+    TRACE_HEADER_KEY = "X-Amzn-Trace-Id"
+
+    KV_PAIR_DELIMITER = ";"
+    KEY_AND_VALUE_DELIMITER = "="
+
+    TRACE_ID_KEY = "Root"
+    TRACE_ID_LENGTH = 35
+    TRACE_ID_VERSION = "1"
+    TRACE_ID_DELIMITER = "-"
+    TRACE_ID_DELIMITER_INDEX_1 = 1
+    TRACE_ID_DELIMITER_INDEX_2 = 10
+    TRACE_ID_FIRST_PART_LENGTH = 8
+
+    PARENT_ID_KEY = "Parent"
+    PARENT_ID_LENGTH = 16
+
+    SAMPLED_FLAG_KEY = "Sampled"
+    SAMPLED_FLAG_LENGTH = 1
+    IS_SAMPLED = "1"
+    NOT_SAMPLED = "0"
+
+    # pylint: disable=too-many-locals
+    # pylint: disable=too-many-return-statements
+    # pylint: disable=too-many-branches
+    # pylint: disable=too-many-statements
+    def extract(
+        self,
+        getter: Getter[TextMapPropagatorT],
+        carrier: TextMapPropagatorT,
+        context: typing.Optional[Context] = None,
+    ) -> Context:
+        trace_header_list = getter(carrier, self.TRACE_HEADER_KEY)
+
+        if not trace_header_list or len(trace_header_list) != 1:
+            return trace.set_span_in_context(
+                trace.INVALID_SPAN, context=context
+            )
+
+        trace_header = trace_header_list[0]
+
+        if not trace_header:
+            return trace.set_span_in_context(
+                trace.INVALID_SPAN, context=context
+            )
+
+        trace_id = trace.INVALID_TRACE_ID
+        span_id = trace.INVALID_SPAN_ID
+        sampled = False
+
+        next_kv_pair_start = 0
+
+        while next_kv_pair_start < len(trace_header):
+            try:
+                kv_pair_delimiter_index = trace_header.index(
+                    self.KV_PAIR_DELIMITER, next_kv_pair_start
+                )
+                kv_pair_subset = trace_header[
+                    next_kv_pair_start:kv_pair_delimiter_index
+                ]
+                next_kv_pair_start = kv_pair_delimiter_index + 1
+            except ValueError:
+                kv_pair_subset = trace_header[next_kv_pair_start:]
+                next_kv_pair_start = len(trace_header)
+
+            stripped_kv_pair = kv_pair_subset.strip()
+
+            try:
+                key_and_value_delimiter_index = stripped_kv_pair.index(
+                    self.KEY_AND_VALUE_DELIMITER
+                )
+            except ValueError:
+                _logger.error(
+                    (
+                        "Error parsing X-Ray trace header. Invalid key value pair: %s. Returning INVALID span context.",
+                        kv_pair_subset,
+                    )
+                )
+                return trace.set_span_in_context(
+                    trace.INVALID_SPAN, context=context
+                )
+
+            value = stripped_kv_pair[key_and_value_delimiter_index + 1 :]
+
+            if stripped_kv_pair.startswith(self.TRACE_ID_KEY):
```

Comment:
this could factored out to it's own method as well: validate_trace_id_format or so on.

Reply:
Will do!